File: C#/Behavioral/ChainOfResponsibility/DesignPatterns.BryanHansen.ChainOfResponsibility/Director.cs
using System;

namespace DesignPatterns.BryanHansen.ChainOfResponsibility
{
    // ConcreteHandler
    public class Director : Handler
    {
        public override void HandleRequest(Request request)
        {
            if(request.IsForAConference())
            {
                Console.WriteLine("Directors can approve conferences.");
            }
            else
            {
                Console.WriteLine($"Director can't approve:\n{request.ToString()}\n");
                Successor.HandleRequest(request);
            }
        }
    }
}

File: Java/src/main/java/behavioral/chainofresponsibility/Director.java
package behavioral.chainofresponsibility;

public class Director extends Handler {

    @Override
    public void handleRequest(Request request) {
        if(request.getRequestType() == RequestType.CONFERENCE) {
            System.out.print("Directors can approve conferences");
        }
        else {
            successor.handleRequest(request);
        }
    }
}
